pkg/bloomcompactor/job.go

```diff
@@ -0,0 +1,43 @@
+package bloomcompactor
+
+import (
+	"github.com/prometheus/common/model"
+
+	"github.com/grafana/loki/pkg/storage/stores/shipper/indexshipper/tsdb/index"
+)
+
+// Job holds a compaction job, which consists of a group of blocks that should be compacted together.
+// Not goroutine safe.
+// TODO: A job should probably contain series or chunks
+type Job struct {
```

Comment:
Nit: I expect we'll want a job to contain multiple series rather than one per fingerprint (fingerprint is roughly analagous to stream). Creating one per fingerprint will likely result in logging millions of streams when the happy path isn't satisfied. That's an inconvenience, but there may be places where it's a performance hit as well.

Instead, we could either
* Create one job for all fingerprints in a TSDB owned by the compactor replica
* Choose some batch-size and create a number of jobs that have that many series|chunks in each (more work but more predictable)

Reply:
That was my original understanding/idea as well. But looking at https://github.com/grafana/loki/pull/11115 looked like we would be listing and processing one series at a time. So I though it was actually easier to filter by the series fingerprint right away.

> - Create one job for all fingerprints in a TSDB owned by the compactor replica
> - Choose some batch-size and create a number of jobs that have that many series|chunks in each (more work but more predictable)

The "problem" I see with this is that we'd need to iterate throughout the whole index to build those batches anyway. And then, iterate again skipping those unowned. So I don't quite see the benefit of creating those batches other than not checking the ring as often (which shouldn't be a problem since`ShuffleShard` ops are cached at the ring level)

> will likely result in logging millions of streams

For each series, the first thing we do is checking if we use the series' FP and skip it if we don't. By logging here I guess you mean the log line _`"skipping job because it is not owned by this shard"`_. I agree this will be too verbose. Since we have a metric to track that (`loki_bloomcompactor_jobs_skipped`), I think we can just remove this log line. Maybe we can even do the same for the tenant?

Reply:
Agreed with Owen to continue with 1 series - 1 job for now. We can easily reevaluate this in future iterations.